nemo/utils/exp_manager.py

```diff
@@ -281,17 +287,25 @@ def exp_manager(trainer: 'pytorch_lightning.Trainer', cfg: Optional[Union[DictCo
     logging.info(f'Experiments will be logged at {log_dir}')
     trainer._default_root_dir = log_dir
 
+    if cfg.log_local_rank_0_only and cfg.log_global_rank_0_only:
+        raise ValueError(
+            f"Cannot set both log_local_rank_0_only and log_global_rank_0_only to True. Please set either one or neither."
+        )
+
     # Handle logging to file
-    if get_envbool(NEMO_ENV_VARNAME_TESTING, False) or world_size <= 32:
-        # If NEMO_TESTING is set (debug mode) or if less than 32 ranks save all log files
+    # Logs on all ranks.
+    if get_envbool(NEMO_ENV_VARNAME_TESTING, False) or (
+        not cfg.log_local_rank_0_only and not cfg.log_global_rank_0_only
```

Comment:
Instead of this not, perform explicit cfg.X is True and cfg.Y is True

Reply:
Fixed.